### ISSUE TITLE
blacklist on scan

### ISSUE THREAD

**Issue by OE1MWW** [egzumer/uv-k5-firmware-custom#459]
On scan, there is sometimes interference (kind of modulated tone) on a repeater channel - different timing and duration, but just strong enough to open the squelch.

It would be desirable to have a temporary 'black list', for example, if the scan stops on this 'interferer', that you press the 'F' button (or another ?) and from now on this channel is skipped during the scan.

**Comment by AubsUK** [comment]
Memory channel scanning or frequency scanning? - I suspect channel, because you mention "repeater channel".

I'm looking at doing some things with memory scanning and this sounds like an interesting option, but that'll be done on a fork of Egzumer's.

**Comment by MicaelJarniac** [comment]
On CHIRP there's a "Skip" column, but I have no idea if it's related to this at all.
![image](https://github.com/egzumer/uv-k5-firmware-custom/assets/19514231/edca4084-a31c-4343-bec5-1d471acfa1e4)

**Comment by zerog2k** [comment]
yes, coming from almost all other kinds of radios, the channel list thing seems like what others (e.g. Yaesu) consider to be scan "banks", but this radio seems to be missing a feature known as "skip", where a memory has a flag on the channel which allows you to enter or cycle through the channel normally, but it is not included in channel scans.

I think this radio is missing this. There are frequencies that I want to have entered into the memory channel, but I don't want them to be part of any scan.
The scan "all" (`*`) is useful to scan _all_ channels, but it's not fully useful without a functional "skip" feature.

I'm studying the existing firmware operation of scanning, and may take this on as a side project to improve this functionality in the future. Because of the way the settings and channels are laid out in memory, and also how CHIRP interacts with the radio memories, It may be a significant departure in functionality, and be a new "fork", which may require "variant" support in CHIRP. (But that also shouldn't be too much of an issue.)

**Comment by prokrypt** [comment]
Check out Armel's fork called F4HWN. Hold M when scan has paused on a station and it's blacklisted until next power-on.
I have my own fork of F4HWN with slightly different features, including the ability to enable monitor mode while in scan mode so you can scroll through the scan list.

**Comment by AubsUK** [comment]
> I think this radio is missing this. There are frequencies that I want to have entered into the memory channel, but I don't want them to be part of any scan.
> The scan "all" (`*`) is useful to scan _all_ channels, but it's not fully useful without a functional "skip" feature.
> 


I've added 10 ScanLists and temporary/permanent lockout to my fork of this firmware which can be configured with the skip in CHIRP or in the radio. Each channel can be in one or more ScanList (useful when certain freqs are used in multiple areas). When scanning, press the ScanList number to enable/disable. When on a broadcasting channel, press * to lockout until power cycle.

https://github.com/AubsUK/uv-k5-firmware-custom
